cmd/dlv/main.go

```diff
@@ -210,17 +211,28 @@ func traceCmd(cmd *cobra.Command, args []string) {
 	status := func() int {
 		const debugname = "debug"
 		var processArgs []string
+		var regexp string
 		if traceAttachPid == 0 {
-			if err := gobuild(debugname); err != nil {
-				return 1
+			var pkg string
+			nargs := args
+			if cmd.ArgsLenAtDash() > 0 {
+				nargs = args[:cmd.ArgsLenAtDash()]
 			}
-			fp, err := filepath.Abs("./" + debugname)
-			if err != nil {
-				fmt.Fprintf(os.Stderr, err.Error())
+			switch len(nargs) {
+			case 1:
+				regexp = args[0]
+			case 2:
+				if cmd.ArgsLenAtDash() != 1 {
+					pkg = args[0]
+				}
+				regexp = args[1]
+			}
+			if err := gobuild(debugname, pkg); err != nil {
 				return 1
 			}
-			defer os.Remove(fp)
+			defer os.Remove("./" + debugname)
```

Comment:
Yeah I agree with that, however I think it's a different issue, different PR. I will send another patch after this lands.